src/mesh_motion/MotionWaves.C

```diff
@@ -0,0 +1,399 @@
+
+#include "mesh_motion/MotionWaves.h"
+
+#include <NaluParsing.h>
+#include "utils/ComputeVectorDivergence.h"
+
+// stk_mesh/base/fem
+#include <stk_mesh/base/FieldBLAS.hpp>
+
+#include <cmath>
+
+namespace sierra {
+namespace nalu {
+
+MotionWaves::MotionWaves(stk::mesh::MetaData& meta, const YAML::Node& node)
+  : MotionBase()
+{
+  load(node);
+
+    // declare divergence of mesh velocity for this motion
+    ScalarFieldType* divV = &(meta.declare_field<ScalarFieldType>(
+      stk::topology::NODE_RANK, "div_mesh_velocity"));
+    stk::mesh::put_field_on_mesh(*divV, meta.universal_part(), nullptr);
+    stk::mesh::field_fill(0.0, *divV);
```

Comment:
Depending on the order of how things are called in Realm, the field data might not be available at this point.